### PR TITLE
Adopt child tasks

### PR DESCRIPTION
**Pull request by lofvir**
@tobiasbrunner Could you please review the changes, in my tests adoption of tasks is working now.

Whats required?
Configure a connection with about 100 CHILD_SA's. While the initiator is processing the QUICK_MODE tasks, trigger an IKE rekeying/reauth on responder side.
The new IKE_SA should adopt the outstanding QUICK_MODE tasks and take over their processing from its predecessor.

### CODE REVIEW

File: src/libcharon/processing/jobs/adopt_children_job.c [hunk 1]
```diff
@@ -127,6 +128,14 @@ METHOD(job_t, execute, job_requeue_t,
 					 * it does trigger an assign_vips(FALSE) event, so we also
 					 * trigger one below */
 					ike_sa->clear_virtual_ips(ike_sa, FALSE);
+
+					/* checkout the new IKE_SA to migrate tasks */
+					new = charon->ike_sa_manager->checkout(charon->ike_sa_manager, this->id);
```
[the remark this answers is not in the document]
Comment (lofvir) [reply src/libcharon/processing/jobs/adopt_children_job.c]: @tobiasbrunner 
I have pushed an alternative approach that should circumvent deadlocking problems.
Could you please review the changes and give some feedback?

**Comment by tobiasbrunner** [comment]
I went with a more generic approach, adding a method to remove any enumerated tasks from queues. Pushed to the _adopt-child-tasks_ branch. Let me know if that works for you.

**Comment by lofvir** [comment]
> I went with a more generic approach, adding a method to remove any enumerated tasks from queues. Pushed to the _adopt-child-tasks_ branch. Let me know if that works for you.

Your approach works likewise and i have tested it successfully for IKEv1.
Have you planned to merge this somehow soon into master?

Thanks for the effort 👍

**Comment by tobiasbrunner** [comment]
You're welcome. I guess it will be included in the next release.

**Pull request closed without merging**